src/routes/safe/components/Apps/utils.ts

```diff
@@ -28,10 +30,10 @@ export const getAppInfoFromOrigin = (origin) => {
   }
 }
 
-export const getAppInfoFromUrl = async (appUrl) => {
+export const getAppInfoFromUrl = async (appUrl: string): Promise<SafeApp> => {
   let res = { id: undefined, url: appUrl, name: 'unknown', iconUrl: appsIconSvg, error: true }
 
-  if (!appUrl) {
+  if (!appUrl || !appUrl.length) {
```

Comment:
if we perform a check here, should type be `string?` instead?

Reply:
Not sure to understand, `appUrl` is `string`.

Reply:
you check if appUrl is defined, but its type is `string` (!appUrl)

if it's possible that it will be undefined then should the type be different or do we need this check at all?

Reply:
Typescript compiler will show us a warning

Reply:
I tested it, and TS does not show any warning. At least running in strictMode = false.

Reply:
<img width="853" alt="Screenshot 2020-05-27 at 18 44 46" src="https://user-images.githubusercontent.com/16622558/83035199-35459c80-a04a-11ea-8a67-740d4f50a4d5.png">

Reply:
haha you are cheating, define it as any first `const someValue: any = undefined;`

Reply:
1. We shouldn't use any
2. I think we're not following each other, my concern is:
The type for `appUrl` is `string` not `string | undefined`, but we're still checking if it's defined. If there's a possibility that it will be undefined, then we should change the type, so typescript compiler will help us to catch and prevent future errors.

Reply:
I think I understand your concern, and you are right.
As we are not using strict checks and somebody could use still `any`, they could send undefined to the function and it could explode.  I would leave the validation, don't hurt, but if it makes you feel uncomfortable, I can remove it.

Reply:
> we are not using strict checks and somebody could use still any

This should not go through a code review 😄 (at lease any for primitive types)

Reply:
validation removed